src/main/java/org/broadinstitute/hellbender/tools/longreads/graph/AlignedBaseGraphCollection.java

```diff
@@ -0,0 +1,583 @@
+     * @param read An aligned {@link GATKRead} object to add to this {@link AlignedBaseGraphCollection}.
+     */
+    public void addSequence(final GATKRead read) {
+
+        final byte[]             bases    = read.getBasesNoCopy();
+        final List<CigarElement> cigar    = read.getCigarElements();
+        final String             contig   = read.getContig();
+        final int                startPos = read.getStart();
+        final String             readName = read.getName();
+
+        addSequence(bases, cigar, contig, startPos, readName);
+    }
+
+    /**
+     * Add in the given aligned sequence information to this {@link AlignedBaseGraphCollection}.
+     * This information is assumed to be pulled directly from a {@link GATKRead} object.  This has implications for
+     * certain parameters (the {@code startPos} in particular).
+     *
+     * Data can only be added to this graph before it is collapsed.  After that, it is an error to add anything else.
+     *
+     * @param bases Bases to be added in the order in which they appear in the read.
+     * @param cigar {@link List<CigarElement>} containing alignment information for each base in {@code bases}.
+     * @param contig The contig to which the given base sequence is aligned.
+     * @param startPos The start position of the first aligned base in the given base sequence.
+     * @param readName The name of the parent read from which this aligned sequence information is derived.
+     */
+    public void addSequence(final byte[] bases,
+                            final List<CigarElement> cigar,
+                            final String contig,
+                            final int startPos,
+                            final String readName) {
+
+        // Make sure we're still OK to add data:
+        if ( isGraphCollapsed ) {
+            throw new GATKException("Attempted to addSequence to " + this.getClass().getName() + " after collapsing it.  This is a logic error.");
+        }
+
+        // Create a queue of nodes that represent the linear graph of all nodes in the given data:
+        final Queue<AlignedBaseVertex> nodeQueue = createAlignedNodes(bases, cigar, contig, startPos, readName);
+
+        if ( nodeQueue.peek() != null ) {
+            final GenomicAndInsertionPosition firstNewNodePos = nodeQueue.peek().getPos();
+
+            // Special handling for empty starting graphs:
+            if ( !contigSubGraphMap.containsKey(contig) ) {
+                // Initialize the graph:
+                initializeGraphWithNodes(contig, nodeQueue);
+            }
+            else {
+                // Merge the given node queue into the existing graph:
+                mergeNodesIntoGraph(contig, nodeQueue);
+            }
+
+            // Collapse nodes in the graph that occur before this sequence if it's time:
+//            if ( contigUncollapsedPositionMap.get(contig).getDistanceTo(firstNewNodePos) > periodicMergeDistance ) {
```

Comment:
Commented code, i'm not sure this check goes here anyway, there should perhaps be a check elsewhere.